### ISSUE TITLE
Pane borders appear as dotted and jagged lines when pane-border-status is set and status bar is at top

### ISSUE THREAD

**Issue by afh** [tmux/tmux#2493]
### Issue description

When setting the pane-border-status while the status bar is on top the pane borders are drawn with wrong characters, i.e. [Box Drawings Light Up Horizontal](https://unicode-table.com/en/#2534) `U+2534` `┴` and [Middle Dot](https://unicode-table.com/en/#00B7) \
`U+00b7` `·` instead of [Box Drawings Light Horizontal](https://unicode-table.com/en/#2500) `U+2500` `─` (see [screenshots below](#screenshots)).

**Steps to reproduce:**
* Build Tmux from the latest git commit (at the time of writing e288ea153ceffe42d88975fc23196c5dc3aeae5d) or use Tmux version `openbsd-6.8`
* Run Tmux `./tmux -vv -f/dev/test -Ltest`
* Configure tmux to use bottom pane-border-status and top status bar as follows:
```
$ ./tmux -vv -f/dev/null -Ltest
$ ./tmux split-window -v
$ ./tmux set pane-border-status bottom
$ ./tmux set status-position top
```

**Expected behaviour:**
The pane borders are drawn as continuous lines using box drawing characters

**Actual behaviour:**
The pane borders are drawn as dotted or "jagged" lines (see [screenshots below](#screenshots)).

**Quick analysis:**
Unfortunately I am too unfamiliar with the tmux codebase to propose a PR. Yet having a cursory look it could be a off-by-one error in [screen-redraw.c:436](https://github.com/tmux/tmux/blob/master/screen-redraw.c#L436) as setting `top` to always be `0` draws the lines correctly. I assume doing so would break Tmux rendering in other ways when `status` is set to a numerical value between `2` and `5` inclusive.

Note that setting `pane-border-status` to `top` will draw all lines with dots and changing `pane-border-lines` to `heavy` or `double` will use the respective heavy or double Box Drawings Up Horizontal character.

<summary id=screenshots>:camera: Screenshots of macOS Terminal.app running Tmux showing pane-border rendering with top and bottom status bar (click image to enlarge).</summary>

|Incorrect pane borders with top status bar|Correct pane borders with bottom status bar|
|---|---|
|<img width="570" alt="tmux-pane-border-rendering-issue" src="https://user-images.githubusercontent.com/16507/101240526-de530b00-36ef-11eb-9a7d-3c11e2323f6b.png">|<img width="570" alt="tmux-pane-border-rendering-issue-status-bottom" src="https://user-images.githubusercontent.com/16507/101240564-32f68600-36f0-11eb-8fba-7982ba43e9ae.png">|

### Required information

```
$ echo $TERM
screen
$ ./tmux -V
tmux next-3.3
$ uname -sp
Darwin i386
$ sw_vers
ProductName:    Mac OS X
ProductVersion: 10.15.7
BuildVersion:   19H15
$ clang --version
Apple clang version 12.0.0 (clang-1200.0.32.27)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

[tmux-client-63319.log](https://github.com/tmux/tmux/files/5646961/tmux-client-63319.log)
[tmux-out-63321.log](https://github.com/tmux/tmux/files/5646962/tmux-out-63321.log)
[tmux-server-63321.log](https://github.com/tmux/tmux/files/5646963/tmux-server-63321.log)

**Comment by nicm** [comment]
Yes, I think this is just a mistake. `screen_redraw_type_of_cell` works on a position in the window not a position on the client tty, so we should not be including the status line and `top` should always be 0. You can see that `screen_redraw_draw_borders_cell` does not include the status line size when it is working out what character to draw, but only adds it on at the very end. I have changed this now in OpenBSD, it will be in GitHub later on. Please let me know if you still see problems. Thanks!

**Comment by afh** [comment]
Thanks @nicm for addressing this so quickly! I hereby confirm that the issue has been fixed with the latest changes to this repository.

Do you think there will be a syspatch(8) available for [OpenBSD 6.8](https://www.openbsd.org/errata68.html)?
